src/exact/ExactMapper.cpp

```diff
@@ -327,13 +327,28 @@ void ExactMapper::map(const Configuration& settings) {
 
 void ExactMapper::coreMappingRoutine(const std::set<unsigned short>& qubitChoice, const CouplingMap& rcm, MappingResults& choiceResults, std::vector<std::vector<std::pair<unsigned short, unsigned short>>>& swaps, long unsigned int limit, unsigned int timeout) {
     const auto& config = results.config;
-
-    // Z3 context
-    context c;
+    using namespace logicbase;
+    // LogicBlock
+    LogicBlock* lb;
+#ifdef Z3_FOUND
+    using namespace z3logic;
+    z3::context  c;
+    z3::solver   slv(c);
+    z3::optimize opt(c);
+    z3::params   p(c);
+    LogicTerm::termType = TermType::BASE;
+    p.set("timeout", timeout);
+    p.set("pb.compile_equality", true);
+    p.set("pp.wcnf", true);
+    p.set("maxres.hill_climb", true);
+    p.set("maxres.pivot_on_correction_set", false);
+    opt.set(p);
+    lb = new Z3LogicOptimizer(c, opt, true);
+#endif
 
     std::vector<unsigned short>                        pi(qubitChoice.begin(), qubitChoice.end());
     unsigned long long                                 piCount{};
-    unsigned long long                                 internalPiCount{};
+    unsigned long long                                 internalPiCount;
```

Comment:
any particular reason for this change? it somehow breaks consistency with how all other values are initialized.

Reply:
this is still to be addressed